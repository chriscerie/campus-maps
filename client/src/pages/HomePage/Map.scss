.map-container {
  height: 92vh;
}
