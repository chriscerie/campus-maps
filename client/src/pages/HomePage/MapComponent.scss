.map-container {
  height: 100vh;
}

.mapboxgl-popup-content {
  top: 3px;
  height: 25px;
  width: 25px;
  border-radius: 50%;
  padding: 10px;
  background: #ddd;
  border: 4px solid white;
}

.mapboxgl-popup-tip {
  border: 12px solid transparent;
  border-top: 17px solid transparent;
}
