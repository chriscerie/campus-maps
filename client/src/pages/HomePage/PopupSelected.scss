.popup-selected-container {
  height: 100%;
  display: flex !important;
}

.popup-selected {
  position: relative;
  height: 40%;
  width: 100%;
  z-index: 10;
  margin-top: auto;
  overflow: auto;
  border-radius: 10px 10px 0 0 !important;
}

.header-title {
  padding: 26px 22px 2px;
  text-align: left;
  font-size: 12px;
  display: flex;
}

.header-title-title {
  margin: 0;
  padding-right: 0.8rem;
}

.header-title-subheader {
  margin: 0;
  font-weight: lighter;
  font-size: 14px;
}

.popup-section {
  padding: 26px 22px;
  text-align: left;
  font-size: 12px;
}

#popup-location-button {
  display: inline-flex;
  box-sizing: border-box;
  justify-content: center;
  font-weight: 500;
  font-size: 1.2em;
  text-decoration: none;
  color: #fff;
  background-color: #2266e3;
  border-radius: 0.5em;
  padding: 0.5em 1em;
  width: 100%;
  margin-top: 0.8em;
  transition: background-color 0.2s ease;
}

#popup-location-button:hover {
  background-color: #1c57c6;
}

@media (min-width: 600px) {
  .popup-selected {
    margin-top: inherit;
    top: 50%;
    height: 90%;
    width: 400px;
    transform: translateY(-50%);
    border-radius: 10px !important;
  }
}
