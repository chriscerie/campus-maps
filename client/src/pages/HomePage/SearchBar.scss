.mapboxgl-ctrl-geocoder {
    z-index : 900;
    position: fixed;
    top: 0;
    left: 0;
}