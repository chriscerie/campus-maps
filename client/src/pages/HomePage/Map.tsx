import { useRef, useEffect } from 'react';
import mapboxgl from 'mapbox-gl';
import './Map.scss';

mapboxgl.accessToken =
  'pk.eyJ1IjoiY2hyaXNjZXJpZSIsImEiOiJja3VvcXBiaGExcG5vMnFtYjhnc3gxcGprIn0.eX9g2ClfVBqYEvecwIPLYw';

function Map() {
  const mapContainerRef = useRef<HTMLDivElement>(null);

  // Initialize map
  useEffect(() => {
    if (!mapContainerRef.current) {
      return;
    }

    const map = new mapboxgl.Map({
      container: mapContainerRef.current,
      // See style options here: https://docs.mapbox.com/api/maps/#styles
      style: 'mapbox://styles/mapbox/streets-v11',
      center: [-104.9876, 39.7405],
      zoom: 12.5,
    });

    // Navigation control (the +/- zoom buttons)
    map.addControl(new mapboxgl.NavigationControl(), 'bottom-right');

    return () => map.remove();
  }, []);

  return <div className="map-container" ref={mapContainerRef} />;
}

export default Map;
