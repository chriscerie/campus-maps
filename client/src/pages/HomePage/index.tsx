import Map from './Map';

function HomePage() {
  return <Map />;
}

export default HomePage;
