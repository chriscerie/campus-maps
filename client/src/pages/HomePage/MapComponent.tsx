import { useRef, useEffect, useState } from 'react';
// Exclude mapboxgl transpilation - https://docs.mapbox.com/mapbox-gl-js/guides/install/#transpiling
// @ts-ignore
// eslint-disable-next-line import/no-webpack-loader-syntax
import mapboxgl from '!mapbox-gl';
import { Map, MapLayerMouseEvent, Popup } from 'mapbox-gl';
import { Feature } from 'geojson';
import PopupSelected from './PopupSelected';
import './MapComponent.scss';

mapboxgl.accessToken =
  'pk.eyJ1IjoiY2hyaXNjZXJpZSIsImEiOiJja3VvcXBiaGExcG5vMnFtYjhnc3gxcGprIn0.eX9g2ClfVBqYEvecwIPLYw';

function MapComponent() {
  const mapContainerRef = useRef<HTMLDivElement>(null);
  const [selected, setSelected] = useState<null | {
    feature: Feature;
    popup: Popup;
  }>(null);

  // Initialize map
  useEffect(() => {
    if (!mapContainerRef.current) {
      return;
    }

    const map: Map = new mapboxgl.Map({
      container: mapContainerRef.current,
      style: 'mapbox://styles/chriscerie/ckuua1bz9it4j18qxt0tyuf71',
      center: [-119.8462, 34.4132],
      zoom: 15.5,
    });

    // Navigation control (the +/- zoom buttons)
    map.addControl(new mapboxgl.NavigationControl(), 'bottom-right');

    // Geolocation control
    map.addControl(
      new mapboxgl.GeolocateControl({
        positionOptions: {
          enableHighAccuracy: true,
        },
        trackUserLocation: true,
        showUserHeading: true,
      }), 'bottom-right'
    );

    // Popup icon when user clicks on a point of interest
    map.on('click', 'poi-label', (e: MapLayerMouseEvent) => {
      if (e.features) {
        // To get name: features.properties.name
        const feature: Feature = e.features[0];

        if (feature.geometry.type === 'Point') {
          const popup: Popup = new mapboxgl.Popup({
            closeButton: false,
          })
            .setLngLat(feature.geometry.coordinates)
            .setHTML('')
            .addTo(map);

          popup.on('close', () => {
            setSelected(null);
          });

          setSelected({
            feature: feature,
            popup: popup,
          });
        }
      }
    });

    map.on('mouseenter', 'poi-label', (_e: MapLayerMouseEvent) => {
      map.getCanvas().style.cursor = 'pointer';
    });

    map.on('mouseleave', 'poi-label', (_e: MapLayerMouseEvent) => {
      map.getCanvas().style.cursor = 'default';
    });

    map.getCanvas().style.cursor = 'default';

    // Start at user location
    if (navigator.geolocation) {
      navigator.geolocation.getCurrentPosition((position) => {
        map.flyTo({
          center: [position.coords.longitude, position.coords.latitude],
        });
      });
    }

    return () => map.remove();
  }, []);

  return (
    <div className="map-container" ref={mapContainerRef}>
      {selected && (
        <PopupSelected
          selected={selected.feature}
          removeSelection={() => {
            selected.popup.remove();
            setSelected(null);
          }}
        />
      )}
    </div>
  );
}

export default MapComponent;
