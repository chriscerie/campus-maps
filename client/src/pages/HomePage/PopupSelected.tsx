import { Divider } from '@mui/material';
import { Box } from '@mui/system';
import { Feature } from 'geojson';
import { Link } from 'react-router-dom';
import './PopupSelected.scss';
import CloseButton from '../../components/CloseButton';
import BottomPopup from '../../components/BottomPopup';

function PopupSelected(props: {
  selected: Feature;
  removeSelection: () => void;
}) {
  const properties = props.selected.properties;

  return (
    properties && (
      <BottomPopup
        zIndex={100}
        bigScreenStyle={{
          marginTop: 'inherit',
          top: '52%',
          height: '85%',
          width: '400px',
          transform: 'translateY(-50%)',
          borderRadius: '10px',
          left: '3em',
        }}
      >
        <Box className="header-title">
          <h1 className="header-title-title">
            {properties.name || 'Name unavailable'}
          </h1>
          <CloseButton
            sx={{ marginLeft: 'auto', marginTop:'0.5rem'}}
            onClick={() => {
              props.removeSelection();
            }}
          />
        </Box>
        <Box className="popup-section" sx={{ paddingTop: 0 }}>
          <h2 className="header-title-subheader">
            {properties.category_en || properties.type}
          </h2>
          <Link to={'/loc/' + props.selected.id} id="popup-location-button">
            More information
          </Link>
        </Box>
        <Divider variant="middle" />
      </BottomPopup>
    )
  );
}

export default PopupSelected;
