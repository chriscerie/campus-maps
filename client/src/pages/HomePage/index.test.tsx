import ShallowRenderer from 'react-test-renderer/shallow';
import HomePage from './index';

describe('HomePage', () => {
  it('should render successfully', () => {
    const renderer = ShallowRenderer.createRenderer();
    const tree = renderer.render(<HomePage />);
    expect(tree).toMatchSnapshot();
  });
});
