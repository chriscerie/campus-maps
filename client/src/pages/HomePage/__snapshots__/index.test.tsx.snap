// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage should render successfully 1`] = `<MapComponent />`;
